File: 2024-08-09-everest-workshop/mars-rover-typescript/rover.js
const RULES = {
  Sf: +1,
  Sb: -1,
  Nf: -1,
  Nb: +1,
};
const DIRECTIONS = ["N", "S"];

function rover({ location, grid, commands }) {
  commands.forEach((command) => {
    if (command == "f" || command == "b")
      move(location, grid, RULES[`${location.direction}${command}`]);
    if (command == "l") location.direction = "N";
  });
  return location;
}

function move(location, grid, amount) {
  location.y += amount;
  if (location.y < 0) {
    location.y = grid.col - 1;
  }
  location.y = location.y % grid.col;
}

module.exports = rover;

File: 2026-06-04-mars-rover-py/rover.py
def rover(position_commands):
    y = position_commands["y"]
    x = position_commands["x"]
    direction = position_commands["direction"]
    if len(position_commands["commands"]) > 0:
        y += 1

    return {
         'x': x,
         'y': y,
         'direction': direction,
     }
